src/Kiota.Builder/Writers/CSharp/CSharpConventionService.cs

```diff
@@ -138,5 +138,26 @@ public override string GetParameterSignature(CodeParameter parameter, CodeElemen
             };
             return $"{parameterType} {parameter.Name.ToFirstCharacterLowerCase()}{defaultValue}";
         }
+
+        public string GetTargetTypeDeclarationRelativeToElement(CodeTypeBase targetType, CodeMethod codeElement)
+        {
+            var targetClass = GetTypeString(targetType, codeElement);
```

Comment:
additional context from the chat with @calebkiage
> Hey, so I think the logic .NET SDK uses for the namespace bug is TranslateTypeAndAvoidUsingNamespaceSegmentNames, but it doesn't seem to do what I need. i.e when type A.B.CType exists and is in scope of a class A.B.C.AType, but in that class I want to use type A.B.C.D.CType, Only referencing it as CType resolves to the wrong type even when I have a using A.B.C.D; statement. In my case, I have to reference it as D.CType for it to properly resolve.

Reply:
Hey @baywet,
For context, we initially used to use the fully qualified name whenever we saw the type name in the namespace string. 

We then found issues with that as outlined [here](https://github.com/microsoft/kiota/pull/1252/files#r812710837) where we fixed that to only use the fully qualified name in the scenarios where the name would be in a different namespace as we saw issues when referencing a type in the same namespace. 

We then had the issue when adding type discriminator support as outlined [here](https://github.com/microsoft/kiota/pull/1174#discussion_r812951249) where the TypeDescriminator functions could collide with type names in same class. 

It looks like this is a different class of problem though. 

@calebkiage I think this functionality can be simplified though to a function that returns a boolean on whether or not to use the full name. The return value can then be used [here](https://github.com/microsoft/kiota/blob/fb7184270759df61f9f08f8b35606cfbed95a788/src/Kiota.Builder/Writers/CSharp/CSharpConventionService.cs#L97) so that `GetTypeString` simply returns the fully qualified string.

Reply:
thanks a lot for that detailed clarification! @calebkiage to update the implementation.

Reply:
Done